activesupport/lib/active_support/core_ext/string/output_safety.rb

```diff
@@ -249,11 +251,51 @@ def #{unsafe_method}!(*args)              # def capitalize!(*args)
       end
     end
 
+    UNSAFE_STRING_METHODS_WITH_BACKREF.each do |unsafe_method|
+      if unsafe_method.respond_to?(unsafe_method)
+        class_eval <<-EOT, __FILE__, __LINE__ + 1
+          def #{unsafe_method}(*args, &block)
```

Comment:
It's nicer if we had a commented code example of what will be produced, like the one we have for the original `UNSAFE_STRING_METHODS`' methods  on the right side:
https://github.com/rails/rails/blob/db0a1257d4aa9b50a613d24eac19d5b19aab717b/activesupport/lib/active_support/core_ext/string/output_safety.rb#L240-L247

Reply:
Thanks for your comment!
I added it (omitted it because my terminal size is 80 columns).

Reply:
Loved the commented code example ! What is the easy way to add the commented code example ? Does any IDE provide such auto generated comments ?

Reply:
[Textbringer](https://github.com/shugo/textbringer), which is an Emacs-like text editor written in Ruby, can generate such comments by the following steps:

1. Put the following code in `~/.textbringer.rb`.

```ruby
def create_binding
  binding
end

define_command :comment_code_example_region do
  |code = read_from_minibuffer("Initialization code: ")|
  env = create_binding
  eval(code, env)
  buffer = Buffer.current
  b, e = buffer.point, buffer.mark
  b, e = e, b if e < b
  buffer.save_excursion do
    buffer.goto_char(e)
    raise "Not at the beginning of a line" if !buffer.beginning_of_line?
  end
  template = buffer.substring(b, e)
  generated = eval("<<~EOT\n" + template + "EOT", env)
  max = template.each_line.map { |l| l.chomp.size }.max
  commented = template.each_line.zip(generated.each_line).map { |tl, cl|
    tl.chomp!
    tl.empty? ? "\n" : tl + " " * (max - tl.size + 2) + "# " + cl
  }.join
  buffer.replace(commented, start: b, end: e)
end
```
2. Start Textbringer.
```
$ textbringer
```
3. Type the following code, or paste by `C-y`.
```ruby
    UNSAFE_STRING_METHODS.each do |unsafe_method|
      if unsafe_method.respond_to?(unsafe_method)
        class_eval <<-EOT, __FILE__, __LINE__ + 1
          def #{unsafe_method}(*args, &block)
            to_str.#{unsafe_method}(*args, &block)
          end

          def #{unsafe_method}!(*args)
            @html_safe = false
            super
          end
        EOT
      end
    end
```
4. Move the cursor to the beginning of the line `def #{unsafe_method}(*args, &block)`.
5. Type `C-SPC`.
6. Move the cursor to the beginning of the line `EOT`.
7. Type `M-x comment_code_example_region RET`.
8. Type `unsafe_method = :capitalize RET`.
9. You will see commented code!

Reply:
👍 Nice. Thank you